Patient Registry System/PatientRegistrySystem.DB/Repos/DbGenericRepository/DbGeneric.cs

```diff
@@ -0,0 +1,249 @@
+                return (List<DomainEntity>)(object)result;
+            }
+            else if (typeof(DbEntity) == typeof(Prescription))
+            {
+                var result = _mapper.Map<List<PrescriptionDto>>(await _applicationIdentityDbContext.Prescription
+                    .AsNoTracking().Where(s => s.PrescriptionId == entityId).ToListAsync());
+                return (List<DomainEntity>)(object)result;
+            }
+            else if (typeof(DbEntity) == typeof(Company))
+            {
+                var result = _mapper.Map<List<CompanyDto>>(await _applicationIdentityDbContext.Company
+                    .AsNoTracking().Where(s => s.CompanyId == entityId).ToListAsync());
+                return (List<DomainEntity>)(object)result;
+            }
+            else if (typeof(DbEntity) == typeof(Medicine))
+            {
+                var result = _mapper.Map<List<MedicineDto>>(await _applicationIdentityDbContext.Medicine
+                    .AsNoTracking().Where(s => s.MedicineId == entityId).ToListAsync());
+                return (List<DomainEntity>)(object)result;
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        public async Task<DomainEntity> FindEntityAsync(int entityId)
+        {
+            if (typeof(DbEntity) == typeof(Record))
+            {
+                var result = _mapper.Map<RecordDto>(await _applicationIdentityDbContext.Record
+                    .AsNoTracking().FirstOrDefaultAsync(s => (s.Status != 1) && (s.ApplicationUserId == entityId)));
+                return (DomainEntity)(object)result;
+            }
+            else if (typeof(DbEntity) == typeof(Prescription))
+            {
+                var result = _mapper.Map<PrescriptionDto>(await _applicationIdentityDbContext.Prescription
+                    .AsNoTracking().FirstOrDefaultAsync(s => s.PrescriptionId == entityId));
+                return (DomainEntity)(object)result;
+            }
+            else if (typeof(DbEntity) == typeof(Company))
+            {
+                var result = _mapper.Map<CompanyDto>(await _applicationIdentityDbContext.Company
+                    .AsNoTracking().FirstOrDefaultAsync(s => s.CompanyId == entityId));
+                return (DomainEntity)(object)result;
+            }
+            else if (typeof(DbEntity) == typeof(Medicine))
+            {
+                var result = _mapper.Map<MedicineDto>(await _applicationIdentityDbContext.Medicine
+                    .AsNoTracking().FirstOrDefaultAsync(s => s.MedicineId == entityId));
+                return (DomainEntity)(object)result;
+            }
+            else
+            {
+                return (DomainEntity)(object)null;
+            }
+        }
+
+        public async Task<List<DomainEntity>> GetAllEntitiesAsync()
+        {
+            if (typeof(DbEntity) == typeof(Record))
```

Comment:
same argument, we need one query to return all records and remove these if statments.